app/models/dialog.rb

```diff
@@ -65,6 +65,14 @@ def validate_children
       errors.add(:base, _("Dialog %{dialog_label} must have at least one Tab") % {:dialog_label => label})
     end
 
+    dialog_field_list = dialog_fields
+    unless dialog_field_list.empty?
```

Comment:
Suggest moving this first `empty?` check into the `check_for_duplicate_dialog_fields` method instead of adding more logic into `validate_children`.

```ruby
...
  dialog_field_list = dialog_fields
  duplicated_field_names = check_for_duplicate_dialog_fields(dialog_field_list)
  unless duplicated_field_names.empty?
    errors.add(:base, _("Dialog field name %{duplicates} cannot be duplicated on a dialog") % {:duplicates => duplicated_field_names})
  end
end

def check_for_duplicate_dialog_fields(dialog_field_list)
  return [] if dialog_field_list.empty?
...
```